week1/index.js

```diff
@@ -1,25 +1,62 @@
-// To run this file: node index.js
-// Then in your browser: http://localhost:8080
+// Determine the port and the initial value of state
+var state = 10;
+var port = 8080;
+
+// Start the HTTP server
 var http = require('http');
+var server = http.createServer(handleRequest);
 
-var port = 8080;
 
-var server = http.createServer();
+// This function is the handler for "response"
+// It sends the requested information to the user
+function handleResponse(response, statusCode, h1Text, pText) {
```

Comment:
Great use of a parameterised function here. This function does one job and only one job, which is very good design.